CNAM-435: Documentation added for extractors

Anfor opened the pull request:
I add documentation to explain how extractors are built and how mcoExtractors works.

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/mco/McoExtractor.scala:
```diff
@@ -15,13 +15,34 @@ trait McoExtractor[EventType <: AnyEvent] extends Extractor[EventType] with McoS
 
   val eventBuilder: EventBuilder
 
+  /** It allows get all need columns from source.
```
I think all this documentation should be put on the Extractor trait.

Anfor replied:
Yes,  It'll  be better placed.

ysebiat commented on Extractors.md:
```diff
@@ -0,0 +1,72 @@
+
+# Extractors
+
+Extractors are a kind of jobs that allows to extract the required columns from the sources. 
```
I think you need to add a sentence after this one, I would suggest the following: 

_and maps them to the Event it is extracting.

 An Extractor is defined by Event type it is extracting and the Source where the information needed to build the extract Event is._

ysebiat commented on Extractors.md:
```diff
@@ -0,0 +1,72 @@
+ 
+Each basic component has a function:
+ - [Extractor](https://github.com/X-DataInitiative/SCALPEL-Extraction/blob/master/src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala)
+ is a trait that works by filtering, extracting and building.It's main method  `extract ` allows to filter the sources and build a dataset of type `Event`.
```
Its instead of it's.

ysebiat reviewed (changes requested):
Some comments.

dsun0720 commented on Extractors.md:
````diff
@@ -0,0 +1,72 @@
+  }
+```
+
+The above elements are responsible for defining the type of `Event` and must implement at least two values and modify them according to their specificity.
````
The values or methods which should be implemented base on their parent trait. I think your words("at least two values") here is a bit confused.

Anfor replied:
ok, this approach is better understood?
"The above elements are responsible for defining the type of `Event` and must implement at least  `columnName` and  `eventBuilder` values and modify them according to their specificity."

dsun0720 replied:
yes

dsun0720 commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
+   */
   def getInput(sources: Sources): DataFrame
 
+  /** It extract columns required to build the event object.
```
"it extracts"

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
 
 trait Extractor[EventType <: AnyEvent] extends Serializable {
 
+  /** It allows to check if the passed row is in the codes set.
```
I think it is better to phrase it as : _Allows to check if the Row from the Source is considered in the current Study._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
 
+  /** It allows to check if the passed row is in the codes set.
+   *
+   * @param codes A set of codes to study.
```
_A set of codes being considered in the Study._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
+   */
   def isInStudy(codes: Set[String])(row: Row): Boolean
 
+  /** It checks if the row is null at columnName.
```
_Checks if the passed Row has the information needed to build the Event._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
+   */
   def isInExtractorScope(row: Row): Boolean
 
+  /** It build an Event from a row.
```
_Builds the Event._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
+   */
   def builder(row: Row): Seq[Event[EventType]]
 
+  /** It allows get all need columns from source.
```
_Gets and prepares all the needed columns from the Source._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/Extractor.scala:
```diff
@@ -9,14 +9,42 @@ import fr.polytechnique.cmap.cnam.etl.sources.Sources
+   */
   def getInput(sources: Sources): DataFrame
 
+  /** It extracts columns required to build the event object.
```
Extracts the Event from the Source.

This function is responsible for glueing different other parts of the Extractor. This method should be considered the unique callable method from a Study perspective.

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/mco/McoExtractor.scala:
```diff
@@ -32,19 +32,43 @@ trait McoExtractor[EventType <: AnyEvent] extends Extractor[EventType] with McoS
     Seq(eventBuilder[EventType](patientId, groupId, code(row), weight, eventDate, endDate))
   }
 
+  /** It gets columnName's value from row.
```
Extracts the tracked value.

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/mco/McoExtractor.scala:
```diff
@@ -32,19 +32,43 @@ trait McoExtractor[EventType <: AnyEvent] extends Extractor[EventType] with McoS
     r.getAs[String](ColNames.PatientID)
   }
 
+  /** It gets EtaNum value, RsaNum value and Year from row.
```
This is wrong description. This is specific to Mco Source and how we group Events from this source together. Better phrasing : 

_Creates an ID that group Events of different categories by concatinating ETA_NUM, RSA_NUM and the YEAR._

ysebiat commented on src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/mco/McoExtractor.scala:
```diff
@@ -32,19 +32,43 @@ trait McoExtractor[EventType <: AnyEvent] extends Extractor[EventType] with McoS
       r.getAs[Int](ColNames.Year).toString
   }
 
+  /** It gets EstimatedStayStart from row.
```
_Extracts the EstimatedStayStart as the start._

You will need to document the EstimatedStayStart method, as this an artefact of our code and does not exist as is in the MCO Source.

ysebiat reviewed (changes requested):
Changes needed in the phrasing.

Anfor committed:
CNAM-435: Documentation added for extractors

CNAM-435: Added Documentation to extractor

CNAM-435: Text changed to better understanding

CNAM-435:Rewrite some documentation to adapt to guidelines.

ysebiat reviewed (approved):
LGTM.
